File: cpp/441.排列硬币.cpp
/*
 * @lc app=leetcode.cn id=441 lang=cpp
 *
 * [441] 排列硬币
 */

// @lc code=start
class Solution {
public:
    int arrangeCoins(int n) {
        if( n < 2 ) return n;
        int i = 1;
        while( n >= 0 )
        {
            n -= i++;
        } 
        return i - 2;
    }
};
// @lc code=end

File: rust/441.排列硬币.rs
/*
 * @lc app=leetcode.cn id=441 lang=rust
 *
 * [441] 排列硬币
 */

// @lc code=start
impl Solution {
    pub fn arrange_coins(n: i32) -> i32 {
        if(n > 1) {
            let mut i = 0;
            let mut sum = n;
            while sum >=  0  {
                i += 1;
                sum -= i;
                //print!("{},{}\n",sum,i);            
            };
            i - 1
        }
        else{
            1
        }
    }
}
// @lc code=end

